PurchasesCoreSwift/Identity/PurchaserInfoManager.swift

```diff
@@ -0,0 +1,191 @@
+        let isCacheStale = deviceCache.isPurchaserInfoCacheStale(appUserID: appUserID,
+                                                                 isAppBackgrounded: isAppBackgrounded)
+        let needsToRefresh = isCacheStale || cachedPurchaserInfo == nil
+        if needsToRefresh {
+            Logger.debug(isAppBackgrounded
+                            ? Strings.purchaserInfo.purchaserinfo_stale_updating_in_background
+                            : Strings.purchaserInfo.purchaserinfo_stale_updating_in_foreground)
+            fetchAndCachePurchaserInfo(appUserID: appUserID,
+                                       isAppBackgrounded: isAppBackgrounded,
+                                       completion: completion)
+            Logger.rcSuccess(Strings.purchaserInfo.purchaserinfo_updated_from_network)
+        } else {
+            if let completion = completion {
+                operationDispatcher.dispatchOnMainThread {
+                    completion(cachedPurchaserInfo, nil)
+                }
+            }
+        }
+    }
+
+    @objc(sendCachedPurchaserInfoIfAvailableForAppUserID:)
+    public func sendCachedPurchaserInfoIfAvailable(appUserID: String) {
+        guard let info = cachedPurchaserInfo(appUserID: appUserID) else {
+            return
+        }
+
+        sendUpdateIfChanged(purchaserInfo: info)
+    }
+
+    @objc(purchaserInfoWithAppUserID:completionBlock:)
+    public func purchaserInfo(appUserID: String, completionBlock: ReceivePurchaserInfoBlock?) {
+        let infoFromCache = cachedPurchaserInfo(appUserID: appUserID)
+        var completionCalled = false
+
+        if let infoFromCache = infoFromCache {
+            Logger.debug(Strings.purchaserInfo.vending_cache)
+            if let completion = completionBlock {
+                completionCalled = true
+                operationDispatcher.dispatchOnMainThread {
+                    completion(infoFromCache, nil)
+                }
+            }
+        }
+
+        // Prevent calling completion twice.
+        let fetchPurchaserInfoCompletion = completionCalled ? nil : completionBlock
+
+        systemInfo.isApplicationBackgrounded { isAppBackgrounded in
+            self.fetchAndCachePurchaserInfoIfStale(appUserID: appUserID,
+                                                   isAppBackgrounded: isAppBackgrounded,
+                                                   completion: fetchPurchaserInfoCompletion)
+        }
+    }
+
+    @objc(cachedPurchaserInfoForAppUserID:)
+    public func cachedPurchaserInfo(appUserID: String) -> PurchaserInfo? {
+        guard let purchaserInfoData = deviceCache.cachedPurchaserInfoData(appUserID: appUserID) else {
+            return nil
+        }
+
+        do {
+            let maybeInfoDict = try JSONSerialization.jsonObject(with: purchaserInfoData) as? [String: Any]
+            guard let purchaserInfoDict = maybeInfoDict, let info = PurchaserInfo(data: purchaserInfoDict) else {
```

Comment:
let's do one `let` per line

Reply:
✅  I'm gonna add this to the style guide.

Reply:
Missed this one